File: LLM-Translations/GPT-4/codenet/Java/Go/s221078931.go

package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

type IntegralRect struct {
	height int
	width  int
}

func (ir IntegralRect) diagonal() int {
	if ir.height >= ir.width {
		return 1<<31 - 1
	}
	return ir.height*ir.height + ir.width*ir.width
}

type ByDiagonalAndHeight []IntegralRect

func (a ByDiagonalAndHeight) Len() int           { return len(a) }
func (a ByDiagonalAndHeight) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
func (a ByDiagonalAndHeight) Less(i, j int) bool {
	d1, d2 := a[i].diagonal(), a[j].diagonal()
	if d1 == d2 {
		return a[i].height < a[j].height
	}
	return d1 < d2
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	rects := make([]IntegralRect, 40000)
	for i := 0; i < 200; i++ {
		for j := 0; j < 200; j++ {
			rects[i*200+j] = IntegralRect{i + 1, j + 1}
		}
	}
	sort.Sort(ByDiagonalAndHeight(rects))

	for {
		scanner.Scan()
		tmpArray := strings.Split(scanner.Text(), " ")

		h, _ := strconv.Atoi(tmpArray[0])
		w, _ := strconv.Atoi(tmpArray[1])

		if h == 0 && w == 0 {
			break
		}

		tmpRect := IntegralRect{h, w}
		for i := 0; i < len(rects); i++ {
			if ByDiagonalAndHeight.Less(rects[i], tmpRect) {
				fmt.Println(rects[i].height, rects[i].width)
				break
			}
		}
	}
}

File: LLM-Translations/GPT-4/codenet/Java/Python/s221078931.py
import sys
from functools import total_ordering

@total_ordering
class IntegralRect:
    def __init__(self, h, w):
        self.height = h
        self.width = w

    def diagonal(self):
        if self.height >= self.width:
            return sys.maxsize
        return self.height * self.height + self.width * self.width

    def __lt__(self, other):
        d1 = self.diagonal()
        d2 = other.diagonal()
        return d1 < d2 or (d1 == d2 and self.height < other.height)

    def __eq__(self, other):
        return self.height == other.height and self.width == other.width

def main():
    rects = [IntegralRect(i + 1, j + 1) for i in range(200) for j in range(200)]
    rects.sort()

    while True:
        h, w = map(int, input().split())

        if h == 0 and w == 0:
            break

        tmp_rect = IntegralRect(h, w)

        for rect in rects:
            if rect > tmp_rect:
                print(rect.height, rect.width)
                break

if __name__ == "__main__":
    main()


